File: php/Program/Clothing.php
<?php
// Clothing.php

// Mengimpor kelas Product.php untuk digunakan dalam definisi kelas Clothing
include 'Product.php';

// Membuat kelas Clothing yang merupakan turunan dari kelas Product
class Clothing extends Product {
    // Mendefinisikan properti-propterti yang spesifik untuk pakaian (ukuran, bahan, jenis kelamin)
    protected $size;
    protected $material;
    protected $gender;

    // Method untuk menetapkan ukuran pakaian
    public function setSize($size) {
        $this->size = $size;
    }

    // Method untuk menetapkan bahan pakaian
    public function setMaterial($material) {
        $this->material = $material;
    }

    // Method untuk menetapkan jenis kelamin pakaian
    public function setGender($gender) {
        $this->gender = $gender;
    }

    // Method untuk mengambil ukuran pakaian
    public function getSize() {
        return $this->size;
    }

    // Method untuk mengambil bahan pakaian
    public function getMaterial() {
        return $this->material;
    }

    // Method untuk mengambil jenis kelamin pakaian
    public function getGender() {
        return $this->gender;
    }
}
?>

File: python/Program/clothing.py
from product import Product  # Impor kelas Product dari modul product
                             # Digunakan untuk melakukan pewarisan dari kelas Product ke kelas Clothing

class Clothing(Product):      # Mendefinisikan kelas Clothing yang merupakan turunan dari kelas Product
    def __init__(self, idProduct, name, brand, price, size, material, gender):
        super().__init__(idProduct, name, brand, price)  # Memanggil konstruktor kelas induk (Product) dengan parameter yang diberikan
        self.size = size         # Inisialisasi atribut size
        self.material = material # Inisialisasi atribut material
        self.gender = gender     # Inisialisasi atribut gender

    def display_info(self):
        # Mengembalikan informasi produk dari kelas Clothing bersama dengan informasi dari kelas Product
        return super().display_info() + [self.size, self.material, self.gender]
